### ISSUE TITLE
Binary caching webdav or S3 backend support

### ISSUE THREAD

**Issue by xiaozhuai** [microsoft/vcpkg#25038]
It would be awesome if vcpkg could support storing binary cache on a webdav or S3 server.
I considered to mount webdav or S3 to filesystem. But there would be many problems.
It will be better if vcpkg support webdav or S3 directly.

**Comment by JackBoosY** [comment]
Yeah, although vcpkg provides binarycaching functionality, we still do not currently provide precompiled binaries.
